### ISSUE TITLE
💅 useAwaitThenable reports await on an intersection with Promise (regression in 2.5.5)

### ISSUE THREAD

**Issue by everton-dgn** [biomejs/biome#11083]
### Environment information

```block
CLI:
  Version:                      2.5.5

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_DISTRIBUTION:           npm
  BIOME_CONFIG_PATH:            unset
  JS_RUNTIME_VERSION:           v24.18.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/11.9.0

Biome Configuration:
  Status:                       Loaded successfully.
  Path:                         biome.json
```

`useAwaitThenable` is enabled as `error`. Same file, same config, only the Biome
version changed: 2.5.2 is clean, 2.5.5 reports. The `rage` output above is from
2.5.5; the 2.5.2 run differs only in the version line.

### Rule name

`nursery/useAwaitThenable`

### Playground link

I could not produce a permalink. The rule is type-aware and I have not verified
whether the playground evaluates type-aware rules, so I did not want to link
something that would look like a clean result for the wrong reason. The
reproduction is a single self-contained file with no imports, so pasting it into
the playground with `useAwaitThenable` enabled should be enough if the playground
does evaluate the rule.

Saving this as `probe.ts` and running `biome lint probe.ts` reproduces it:

```ts
function plainPromise(): Promise<number> {
  return Promise.resolve(1)
}

function intersectionPromise(): Promise<number> & { cancel: () => void } {
  const promise = Promise.resolve(1) as Promise<number> & {
    cancel: () => void
  }
  promise.cancel = () => {}
  return promise
}

export async function probe(): Promise<number> {
  const a = await plainPromise()
  const b = await intersectionPromise()
  return a + b
}
```

On 2.5.5, line 15 (`await intersectionPromise()`) is reported:

```block
lint/nursery/useAwaitThenable
  > 15 │   const b = await intersectionPromise()
```

Line 14 (`await plainPromise()`) is not reported, so the trigger is the
intersection and not the `await` itself. On 2.5.2 the same file produces no
`useAwaitThenable` diagnostic.

`Promise<T> & { ... }` is a real promise: the intersection only attaches extra
members. The pattern shows up whenever a helper returns a promise carrying a
method, for example a cancellable wait:

```ts
export function waitForSignal<T>(
  timeoutMs: number,
  setup: (resolve: (value: T) => void, reject: (error: Error) => void) => () => void
): Promise<T> & { cancel: () => void }
```

Removing the `await` to satisfy the rule would break the code, so on 2.5.5 the
only options are suppressing every call site or disabling the rule. It is what
kept us on 2.5.2.

This may share a root cause with the closed #8476 (`useAwaitThenable` false
positives with drizzle), whose query builders are also promises with extra
members, though that report was diagnosed as thenables needing `await` or
`.execute` unwrapping, which is a different shape from a plain intersection with
`Promise<T>`.

### Expected result

`await` on a value whose type is an intersection that includes `Promise<T>`
should not be reported, since the value is a promise. The intersection members
alongside `Promise<T>` should not affect thenability.

**Comment by ematipico** [comment]
I understand that you didn't want to use the playground, but we still need a reproduction. The playground enables type-aware rules.

**Comment by biome-cookie** [comment]
Hello @everton-dgn, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by everton-dgn** [comment]
Minimal reproduction repository: https://github.com/everton-dgn/biome-use-await-thenable-intersection-repro

It is pinned to Biome 2.5.5. Running:

```sh
pnpm install
pnpm lint
```

reports the second `await`, whose operand has type `Promise<number> & { cancel: () => void }`. The first `await`, whose operand is a plain `Promise<number>`, is not reported.

The same source and configuration are clean on Biome 2.5.2:

```sh
pnpm dlx @biomejs/biome@2.5.2 lint src/index.ts
```

The current playground build is clean for this case, so the repository pins the affected release.

**Comment by ematipico** [comment]
This is fixed in v2.5.6
